### PR TITLE
added configurable data/etc dir and options validating config files only

### PR DESCRIPTION
**Pull request by edisonguo**
@bje- The changes in this PR essentially come from Chris's deployment feedbacks.

### CODE REVIEW

File: ows.go [hunk 1]
```diff
@@ -39,7 +39,10 @@ import (
 var configMap map[string]*utils.Config
 
 var (
-	port = flag.Int("p", 8080, "Server listening port.")
+	port            = flag.Int("p", 8080, "Server listening port.")
+	serverDataDir   = flag.String("data_dir", utils.DataDir, "Server data directory.")
```
Comment (bje-) [inline ows.go]: Why is this needed when you can pass `--datarootdir` and `--sysconfdir` to `configure`?

File: ows.go [hunk 2]
```diff
@@ -39,7 +39,10 @@ import (
...
+	serverConfigDir = flag.String("conf_dir", utils.EtcDir, "Server config directory.")
```
Comment (bje-) [inline ows.go]: Likewise.

**Comment by edisonguo** [comment]
We need these because the build environment and the production environment that actually runs `ows` can have different directory structures. What happens is that there is a build machine that builds gsky and then `tar` all the exectuables. Then the tarball gets deployed by `puppet` scripts to dozens of production machines whose data and etc directory locations/structures don't exactly mirror the ones in the build environment for various production reasons that that dev doesn't care about.

File: utils/config.go [hunk 3]
```diff
@@ -332,6 +333,40 @@ func LoadAllConfigFiles(rootDir string) (map[string]*Config, error) {
 	return configMap, err
 }
 
+//https://github.com/hashicorp/packer/blob/master/common/json/unmarshal.go
```
Comment (bje-) [inline utils/config.go]: Unit test case needed for this new function.

File: ows.go [hunk 4]
```diff
@@ -39,7 +39,8 @@ import (
 var configMap map[string]*utils.Config
 
 var (
-	port = flag.Int("p", 8080, "Server listening port.")
+	port           = flag.Int("p", 8080, "Server listening port.")
```
Comment (bje-) [inline ows.go]: Don't forget to update any documentation. If there is any. ;-)
